### PR TITLE
Ziafazal/changes after birch rebase

### PR DESCRIPTION
**Pull request by ziafazal**
@antoviaque @e-kolpakov @smagoun Created this new PR to reflect only those changes I made after rebase. I hope now it'll make reviewing easy.

### CODE REVIEW

**Comment by e-kolpakov** [comment]
@ziafazal code changes look good, except some requirements/edx/custom.txt records are missing - I'll send a PR shortly to add them back.

**Comment by smagoun** [comment]
@mattdrayer This is the PR for the Birch rebase that contains only our changes. Please take a look

File: common/lib/xmodule/xmodule/x_module.py [hunk 1]
```diff
@@ -1480,7 +1480,7 @@ def get_course_template_context(self):
         user_cohort_id = get_cohort_id(user, course_id)
 
         unsafethreads, query_params = get_threads(request, course_id)
-        threads = [utils.safe_content(thread, course_id) for thread in unsafethreads]
+        threads = [utils.prepare_content(thread, course_id) for thread in unsafethreads]
```
Comment (mattdrayer) [inline common/lib/xmodule/xmodule/x_module.py]: Is this change due to a rename, or are we now using an alternative workflow?

File: lms/djangoapps/gradebook/management/commands/tests/test_generate_gradebook_entries.py [hunk 2]
```diff
@@ -29,7 +33,7 @@ class GenerateGradebookEntriesTests(TestCase):
     def setUp(self):
 
         # Turn off the signalling mechanism temporarily
-        settings._wrapped.default_settings.FEATURES['SIGNAL_ON_SCORE_CHANGED'] = False
+        #settings._wrapped.default_settings.FEATURES['SIGNAL_ON_SCORE_CHANGED'] = False
```
Comment (mattdrayer) [inline lms/djangoapps/gradebook/management/commands/tests/test_generate_gradebook_entries.py]: This line has been commented out -- is it intended to stay off?  Should it just be removed?

File: lms/djangoapps/gradebook/management/commands/tests/test_generate_gradebook_entries.py [hunk 3]
```diff
@@ -135,6 +139,7 @@ def get_module_for_user(self, user, course, problem):
             course.id
         )._xmodule
 
+    @skip
```
Comment (mattdrayer) [inline lms/djangoapps/gradebook/management/commands/tests/test_generate_gradebook_entries.py]: Are we skipping this test for any particular reason?

File: lms/djangoapps/organizations/management/commands/tests/test_move_organizations_entries.py [hunk 4]
```diff
@@ -40,7 +41,7 @@ def setUp(self):
             org.groups.add(group1, group2)
             org.workgroups.add(workgroup)
 
-
+    @skip
```
Comment (mattdrayer) [inline lms/djangoapps/organizations/management/commands/tests/test_move_organizations_entries.py]: If this test is no longer valid/applicable, then I would say we should just remove it, along with the script that it was validating -- thoughts?

File: lms/djangoapps/projects/management/commands/tests/test_migrate_project_courseids_v2.py [hunk 5]
```diff
@@ -5,19 +5,21 @@
 from datetime import datetime
 import uuid
 
+from django.conf import settings
 from django.contrib.auth.models import Group, User
 from django.test import TestCase
```
Comment (mattdrayer) [inline lms/djangoapps/projects/management/commands/tests/test_migrate_project_courseids_v2.py]: Do you still need TestCase here?  Be sure to execute both 'paver run_quality' and 'paver run_pep8' in your devstack.

File: lms/djangoapps/projects/management/commands/tests/test_migrate_project_courseids.py [hunk 6]
```diff
@@ -5,19 +5,23 @@
 from datetime import datetime
 import uuid
 
+from django.conf import settings
 from django.contrib.auth.models import Group, User
 from django.test import TestCase
```
Comment (mattdrayer) [inline lms/djangoapps/projects/management/commands/tests/test_migrate_project_courseids.py]: Do you still need TestCase?

File: lms/djangoapps/gradebook/tests.py [hunk 7]
```diff
@@ -16,17 +16,20 @@
 from capa.tests.response_xml_factory import StringResponseXMLFactory
 from courseware import module_render
 from courseware.model_data import FieldDataCache
-from courseware.tests.modulestore_config import TEST_DATA_MIXED_MODULESTORE
 from student.tests.factories import UserFactory, AdminFactory
 from courseware.tests.factories import StaffFactory
 from xmodule.modulestore.tests.factories import CourseFactory, ItemFactory
+from xmodule.modulestore.tests.django_utils import ModuleStoreTestCase, mixed_store_config
```
Comment (mattdrayer) [inline lms/djangoapps/gradebook/tests.py]: Do you still need TestCase above?

File: lms/djangoapps/gradebook/management/commands/tests/test_migrate_gradebook_courseids_v2.py [hunk 8]
```diff
@@ -5,19 +5,22 @@
 from datetime import datetime
 import uuid
 
+from django.conf import settings
```
Comment (mattdrayer) [inline lms/djangoapps/gradebook/management/commands/tests/test_migrate_gradebook_courseids_v2.py]: Looks like TestCase (below) can go?

File: lms/djangoapps/projects/tests/test_peer_reviews.py [hunk 9]
```diff
@@ -8,17 +8,19 @@
...
 from django.contrib.auth.models import User
 from django.core.cache import cache
 from django.test import TestCase, Client
 from django.test.utils import override_settings
 
-from courseware.tests.modulestore_config import TEST_DATA_MIXED_MODULESTORE
 from projects.models import Project, Workgroup
 from student.models import anonymous_id_for_user
 from xmodule.modulestore.tests.factories import CourseFactory, ItemFactory
+from xmodule.modulestore.tests.django_utils import ModuleStoreTestCase, mixed_store_config
```
Comment (mattdrayer) [inline lms/djangoapps/projects/tests/test_peer_reviews.py]: TestCase import above can be removed

File: lms/djangoapps/projects/tests/test_receivers.py [hunk 10]
```diff
@@ -6,20 +6,22 @@
...
 import uuid
 
+from django.conf import settings
 from django.contrib.auth.models import User
 from django.test import TestCase
 from django.test.utils import override_settings
 
-from courseware.tests.modulestore_config import TEST_DATA_MIXED_MODULESTORE
 from xmodule.modulestore.tests.factories import CourseFactory, ItemFactory
+from xmodule.modulestore.tests.django_utils import ModuleStoreTestCase, mixed_store_config
```
Comment (mattdrayer) [inline lms/djangoapps/projects/tests/test_receivers.py]: TestCase?

File: lms/djangoapps/projects/tests/test_submission_reviews.py [hunk 11]
```diff
@@ -8,16 +8,18 @@
...
 from django.contrib.auth.models import User
 from django.core.cache import cache
 from django.test import TestCase, Client
 from django.test.utils import override_settings
 
-from courseware.tests.modulestore_config import TEST_DATA_MIXED_MODULESTORE
 from projects.models import Project, Workgroup, WorkgroupSubmission
 from student.models import anonymous_id_for_user
 from xmodule.modulestore.tests.factories import CourseFactory, ItemFactory
+from xmodule.modulestore.tests.django_utils import ModuleStoreTestCase, mixed_store_config
```
Comment (mattdrayer) [inline lms/djangoapps/projects/tests/test_submission_reviews.py]: Unused TestCase import above

File: lms/djangoapps/projects/tests/test_workgroup_reviews.py [hunk 12]
```diff
@@ -8,16 +8,18 @@
...
 from django.contrib.auth.models import User
 from django.core.cache import cache
 from django.test import TestCase, Client
 from django.test.utils import override_settings
 
-from courseware.tests.modulestore_config import TEST_DATA_MIXED_MODULESTORE
 from projects.models import Project, Workgroup, WorkgroupSubmission
 from student.models import anonymous_id_for_user
 from xmodule.modulestore.tests.factories import CourseFactory, ItemFactory
+from xmodule.modulestore.tests.django_utils import ModuleStoreTestCase, mixed_store_config
```
Comment (mattdrayer) [inline lms/djangoapps/projects/tests/test_workgroup_reviews.py]: TestCase import can be removed

File: lms/djangoapps/projects/tests/test_workgroups.py [hunk 13]
```diff
@@ -9,20 +9,26 @@
...
 from api_manager.models import GroupProfile
-from courseware.tests.modulestore_config import TEST_DATA_MIXED_MODULESTORE
 from projects.models import Project, Workgroup
 from student.tests.factories import CourseEnrollmentFactory
 from xmodule.modulestore.tests.factories import CourseFactory, ItemFactory
-from course_groups.cohorts import (get_cohort_by_name, remove_user_from_cohort,
-                                   delete_empty_cohort, is_user_in_cohort, get_course_cohort_names)
-from course_groups.models import CourseUserGroup
-
+from xmodule.modulestore.tests.django_utils import ModuleStoreTestCase, mixed_store_config
```
Comment (mattdrayer) [inline lms/djangoapps/projects/tests/test_workgroups.py]: TestCase again

**Comment by mattdrayer** [comment]
@ziafazal -- finished my first pass -- overall the changeset LGTM -- let me know when you're done making changes and I'll take another look.

File: common/lib/xmodule/xmodule/x_module.py (hunk 1 above)
Comment (ziafazal) [inline common/lib/xmodule/xmodule/x_module.py]: `safe_content` method has been replaced with `prepare_content` in cohorts app.

File: lms/djangoapps/gradebook/management/commands/tests/test_generate_gradebook_entries.py (hunk 2 above)
Comment (ziafazal) [inline lms/djangoapps/gradebook/management/commands/tests/test_generate_gradebook_entries.py]: Yeah that was the reason for one of the test in gradebook was failing. uncommenting it.

File: lms/djangoapps/gradebook/management/commands/tests/test_generate_gradebook_entries.py (hunk 3 above)
Comment (ziafazal) [inline lms/djangoapps/gradebook/management/commands/tests/test_generate_gradebook_entries.py]: No need to skip now.

File: lms/djangoapps/organizations/management/commands/tests/test_move_organizations_entries.py (hunk 4 above)
Comment (ziafazal) [inline lms/djangoapps/organizations/management/commands/tests/test_move_organizations_entries.py]: Yeah this test is no longer applicable. It is related to moving data from lms/api_manager/organizations to lms/organizations app. Removing this.

**Comment by ziafazal** [comment]
@mattdrayer I have made few changes as per your feedback. Could you please take another look?

**Comment by e-kolpakov** [comment]
@ziafazal looks like you have already updated ibirch-rebase to contain Notifications update, so this PR is now outdated. Do you plan to maintain it, or post new one?

**Comment by ziafazal** [comment]
@e-kolpakov No i have not rebased  ibirch-rebase with master to get notifications change set.

**Comment by e-kolpakov** [comment]
@ziafazal oh, sorry, for some reason I thought much less files were modified in this PR. Anyway, do I understand right that notifications changeset is planned to be included in birch rebase?

**Comment by ziafazal** [comment]
@e-kolpakov yeah I'm rebasing `ziafazal/ibirch-rebase` branch with `master` to get notifications related changes and then I'm planning to cherry-pick commits in this branch.

**Comment by chrisndodge** [comment]
I'll look at ziafazal/ibirch-rebase to do an eye inspection regarding Notifications.

**Comment by ziafazal** [comment]
@e-kolpakov @mattdrayer closing this since I have cherry-picked commits in this PR into Ziafazal/ibirch rebase branch.

**Pull request closed without merging**